data/card_definitions.json

```diff
@@ -6485,6 +6485,383 @@
 		"NOTEEND": "NAGO CARDS END"
 	},
 
+	{
+		"NOTE": "NOEL CARDS START",
+		"id": "noel_valkyrieveil",
+		"type": "ultra",
+		"display_name": "Valkyrie Veil",
+		"force_cost": 0,
+		"gauge_cost": 3,
+		"range_min": 1,
+		"range_max": 2,
+		"speed": 2,
+		"power": 3,
+		"armor": 5,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"effect_type": "stun_immunity"
+			},
+			{
+				"timing": "before",
+				"condition": "was_hit",
+				"effect_type": "powerup",
+				"amount": 4
+			},
+			{
+				"timing": "before",
```

Comment:
It may be preferable to make this effect an "and" field under the above; that way it won't need to ask the player which one they want to resolve first (plus you won't need to repeat the timing and condition)